forestry/api/genetics/IMutationCondition.java

```diff
@@ -7,14 +7,16 @@
 
 import net.minecraft.world.World;
 
+import forestry.api.core.IClimateProvider;
+
 public interface IMutationCondition {
 
 	/**
 	 * Returns a float from 0 to 1 representing the chance for mutation to occur.
 	 * Most will return 1 if the condition is met and 0 otherwise,
 	 * but the float offers flexibility for more advanced conditions.
 	 */
-	float getChance(World world, int x, int y, int z, IAllele allele0, IAllele allele1, IGenome genome0, IGenome genome1);
+	float getChance(World world, int x, int y, int z, IAllele allele0, IAllele allele1, IGenome genome0, IGenome genome1, IClimateProvider climate);

```

Comment:
Rather than breaking backward compatibility with changing method signature. Why not overloading method with new argument and keep the original method without the `IClimateProvider`?

Reply:
Keeping the original method signature is one of the options to help improve safety, but we ultimately need to delete the old method signature. The issue is this bug:
 * https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7297 (namely, bee mutations do not take into account environmental modifications in an alveary)

It is not possible to fix that bug with the old method signature, so any implementation with the old method signature will necessarily have that bug.